### PR TITLE
Refactor and Fix Stationloving into Shiploving

### PR DESCRIPTION
**Pull request by ZephyrTFA**
Unit tests make me sad.

### CODE REVIEW

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 1]
```diff
@@ -621,7 +621,7 @@ This is here to make the tiles around the station mininuke change when it's arme
 
 /obj/item/disk/nuclear/ComponentInitialize()
 	. = ..()
-	AddComponent(/datum/component/stationloving, !fake)
+	AddComponent(/datum/component/shiploving, !fake)
```
Comment (MarkSuckerberg) [inline code/modules/antagonists/nukeop/equipment/nuclearbomb.dm]:
```suggestion
	AddComponent(/datum/component/shiploving, inform_admins = !fake)
```

File: code/modules/antagonists/blob/structures/core.dm [hunk 2]
```diff
@@ -68,4 +68,4 @@
 
 /obj/structure/blob/core/ComponentInitialize()
 	. = ..()
-	AddComponent(/datum/component/stationloving, FALSE, TRUE)
+	AddComponent(/datum/component/shiploving, FALSE, TRUE)
```
Comment (MarkSuckerberg) [inline code/modules/antagonists/blob/structures/core.dm]:
```suggestion
	AddComponent(/datum/component/shiploving, inform_admins = FALSE, allow_death = TRUE)
```

File: code/datums/components/shiploving.dm [hunk 3]
```diff
@@ -14,34 +15,41 @@
 	RegisterSignal(parent, list(COMSIG_ITEM_MARK_RETRIEVAL), .proc/check_mark_retrieval)
 	src.inform_admins = inform_admins
 	src.allow_death = allow_death
+	for(var/obj/structure/overmap/ship/simulated/ship as anything in SSovermap.simulated_ships)
+		if(get_area(parent) in ship.shuttle.shuttle_areas)
+			parent_ship = ship
+			break
+	if(!parent_ship)
+		message_admins("[parent] created with [type] outside a ship, assigning a ship at random!")
+		parent_ship = pick(SSovermap.simulated_ships)
```
Comment (MarkSuckerberg) [inline code/datums/components/shiploving.dm]:
```suggestion
	if(!parent_ship)
		  for(var/obj/structure/overmap/ship/simulated/ship as anything in SSovermap.simulated_ships)
			  if(get_area(parent) in ship.shuttle.shuttle_areas)
				  parent_ship = ship
				  break
	if(!parent_ship)
		message_admins("[parent] created with [type] outside a ship, assigning a ship at random!")
		parent_ship = pick(SSovermap.simulated_ships)
```

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]

File: code/datums/components/shiploving.dm [hunk 4]
```diff
@@ -14,34 +15,41 @@
 	RegisterSignal(parent, list(COMSIG_ITEM_MARK_RETRIEVAL), .proc/check_mark_retrieval)
 	src.inform_admins = inform_admins
 	src.allow_death = allow_death
+	for(var/obj/structure/overmap/ship/simulated/ship as anything in SSovermap.simulated_ships)
```
Comment (ZephyrTFA) [inline code/datums/components/shiploving.dm]:
```suggestion
	src.allow_death = allow_death
	src.parent_ship = parent_ship
	for(var/obj/structure/overmap/ship/simulated/ship as anything in SSovermap.simulated_ships)
```

File: code/datums/components/shiploving.dm [hunk 5]
```diff
@@ -14,34 +15,41 @@
...
+		parent_ship = pick(SSovermap.simulated_ships)
 	check_in_bounds() // Just in case something is being created outside of station/centcom
 
-/datum/component/stationloving/InheritComponent(datum/component/stationloving/newc, original, inform_admins, allow_death)
-	if (original)
+/datum/component/shiploving/InheritComponent(datum/component/shiploving/newc, i_am_original, inform_admins, allow_death)
+	if (i_am_original)
 		if (newc)
 			inform_admins = newc.inform_admins
 			allow_death = newc.allow_death
```
Comment (ZephyrTFA) [inline code/datums/components/shiploving.dm:33]:
```suggestion
			allow_death = newc.allow_death
			parent_ship = newc.parent_ship
```

File: code/datums/components/shiploving.dm [hunk 6]
```diff
@@ -14,34 +15,43 @@
...
+				  parent_ship = ship
+				  break
+	if(!parent_ship)
+		message_admins("[parent] created with [type] outside a ship, assigning a ship at random!")
+		parent_ship = pick(SSovermap.simulated_ships)
 	check_in_bounds() // Just in case something is being created outside of station/centcom
 
-/datum/component/stationloving/InheritComponent(datum/component/stationloving/newc, original, inform_admins, allow_death)
-	if (original)
+/datum/component/shiploving/InheritComponent(datum/component/shiploving/newc, i_am_original, inform_admins, allow_death)
```
Comment (ZephyrTFA) [inline code/datums/components/shiploving.dm]:
```suggestion
/datum/component/shiploving/InheritComponent(datum/component/shiploving/newc, i_am_original, parent_ship, inform_admins, allow_death)
```

**Comment by ZephyrTFA** [comment]
@MarkSuckerberg I can't merge this because I dont want to setup the github cli to override automerge

**Review by MarkSuckerberg — APPROVED** [review]

**Pull request merged**